libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Network/Network/Queue/CSFNetwork.m

```diff
@@ -417,6 +417,7 @@ - (void)delegate_networkCanceledAction:(CSFAction*)action {
             [delegate network:self didCancelAction:action];
         }
     }];
+    [action dequeueNetwork:self];
```

Comment:
this delegate_networkCompletedAction gets called after completeOperationWithError: is called.  So essentially it would be same thing right.

You think it makes more sense for CSFAction to own dequeueing?